### ISSUE TITLE
UX throws 500s with: can't evaluate field ReleaseVersion in type *public.VersionInfo"

### ISSUE THREAD

**Issue by sokoow** [linkerd/linkerd2#1702]
Hey All,

I deployed LD2 with linkerd install, all seemed to deploy well, but when I try to open up the UX part, I constantly get: 

```time="2018-09-24T13:32:00Z" level=error msg="Unexpected API response: 500 Internal Server Error"
time="2018-09-24T13:32:00Z" level=error msg="template: app.tmpl.html:3:33: executing \"content\" at <.Data.ReleaseVersion>: can't evaluate field ReleaseVersion in type *public.VersionInfo"
2018/09/24 13:32:00 http: multiple response.WriteHeader calls```

Checked API pods for errors, nothing there corresponding to this. Any idea?

**Comment by dadjeibaah** [comment]
@sokoow thanks for filing this issue, what do you get when you run `linkerd check` a couple of minutes after install? When you look that the logs for the linkerd web container, i.e. `kubectl -n linkerd logs web-xxxxx-xxx` what do you get?

**Comment by sokoow** [comment]
```$ linkerd check
kubernetes-api: can initialize the client..................................[ok]
kubernetes-api: can query the Kubernetes API...............................[ok]
kubernetes-api: is running the minimum Kubernetes API version..............[ok]
linkerd-api: control plane namespace exists................................[ok]
linkerd-api: control plane pods are ready..................................[ok]
linkerd-api: can initialize the client.....................................[ok]
linkerd-api: can query the control plane API...............................[ok]
linkerd-api[kubernetes]: control plane can talk to Kubernetes..............[ok]
linkerd-api[prometheus]: control plane can talk to Prometheus..............[ok]
linkerd-version: can determine the latest version..........................[ok]
linkerd-version: cli is up-to-date.........................................[ok]
linkerd-version: control plane is up-to-date...............................[ok]

Status check results are [ok]```

```$ kubectl logs web-866fdc6b9c-p2hf6 -n linkerd -c web
time="2018-09-24T10:47:31Z" level=info msg="running version stable-2.0.0"
time="2018-09-24T10:47:31Z" level=info msg="starting admin server on :9994"
time="2018-09-24T10:47:31Z" level=info msg="starting HTTP server on :8084"
time="2018-09-24T13:29:57Z" level=error msg="Unexpected API response: 500 Internal Server Error"
time="2018-09-24T13:29:57Z" level=error msg="template: app.tmpl.html:3:33: executing \"content\" at <.Data.ReleaseVersion>: can't evaluate field ReleaseVersion in type *public.VersionInfo"
2018/09/24 13:29:57 http: multiple response.WriteHeader calls
time="2018-09-24T13:30:10Z" level=error msg="Unexpected API response: 500 Internal Server Error"
time="2018-09-24T13:30:10Z" level=error msg="template: app.tmpl.html:3:33: executing \"content\" at <.Data.ReleaseVersion>: can't evaluate field ReleaseVersion in type *public.VersionInfo"
2018/09/24 13:30:10 http: multiple response.WriteHeader calls
time="2018-09-24T13:32:00Z" level=error msg="Unexpected API response: 500 Internal Server Error"
time="2018-09-24T13:32:00Z" level=error msg="template: app.tmpl.html:3:33: executing \"content\" at <.Data.ReleaseVersion>: can't evaluate field ReleaseVersion in type *public.VersionInfo"
2018/09/24 13:32:00 http: multiple response.WriteHeader calls
time="2018-09-24T13:36:11Z" level=error msg="Unexpected API response: 500 Internal Server Error"
time="2018-09-24T13:36:11Z" level=error msg="template: app.tmpl.html:3:33: executing \"content\" at <.Data.ReleaseVersion>: can't evaluate field ReleaseVersion in type *public.VersionInfo"
2018/09/24 13:36:11 http: multiple response.WriteHeader calls
time="2018-09-24T13:39:15Z" level=error msg="Unexpected API response: 500 Internal Server Error"
time="2018-09-24T13:39:15Z" level=error msg="template: app.tmpl.html:3:33: executing \"content\" at <.Data.ReleaseVersion>: can't evaluate field ReleaseVersion in type *public.VersionInfo"
2018/09/24 13:39:15 http: multiple response.WriteHeader calls
time="2018-09-24T13:42:01Z" level=error msg="Unexpected API response: 500 Internal Server Error"
time="2018-09-24T13:42:01Z" level=error msg="template: app.tmpl.html:3:33: executing \"content\" at <.Data.ReleaseVersion>: can't evaluate field ReleaseVersion in type *public.VersionInfo"
2018/09/24 13:42:01 http: multiple response.WriteHeader calls```

**Comment by dadjeibaah** [comment]
Strange, is this a fresh install of Linkerd or did you already have a Linkerd control plane already running before you install `stable-2.0.0`?

**Comment by sokoow** [comment]
Well, I just did linkerd install and that's it...

**Comment by klingerf** [comment]
@sokoow Thanks for reporting this -- it's definitely unexpected. It looks like there are a couple of things happening here.

1. The call from the web service to the public-api is failing. You should verify that the web pod can talk to the public API. If you exec into the web pod, you should be able to run:

```
curl -vX POST -o /dev/null api.linkerd.svc.cluster.local:8085/api/v1/Version
```

And that should return an HTTP 200 response. If the request does not succeed, it's possible you have some sort of network configuration that's preventing the web pod from querying the public-api service.

2. Our error handling in this situation has some issues. We're double rendering in the event of a template error, and that's preventing us from displaying a meaningful error in the web UI when this type of failure is encountered. I can post a PR soon that will hopefully help to surface the error better when this type of failure is encountered.

**Comment by klingerf** [comment]
I fixed the web rendering in #1710 so that we should properly surface the error in the web UI, which will go out with the next release. @sokoow, and luck curling the public API from the web pod as described in my previous comment?

**Comment by sokoow** [comment]
confirmed and working now
